### ISSUE TITLE
Losing audio track following a webRTC Glare detection 

### ISSUE THREAD

**Issue by EdGallimore** [twilio/twilio-video.js#1508]
- [X] I have verified that the issue occurs with the latest twilio-video.js release and is not marked as a known issue in the [CHANGELOG.md](https://github.com/twilio/twilio-video.js/blob/master/CHANGELOG.md).
 - [X] I reviewed the [Common Issues](https://github.com/twilio/twilio-video.js/blob/master/COMMON_ISSUES.md) and open GitHub issues and verified that this report represents a potentially new issue.
 - [X] I verified that the [Quickstart](https://github.com/twilio/video-quickstart-js) application works in my environment.
 - [X] I am not sharing any [Personally Identifiable Information (PII)](https://www.twilio.com/docs/glossary/what-is-personally-identifiable-information-pii)
       or sensitive account information (API keys, credentials, etc.) when reporting this issue.

**Context:**
Two parties are contacting each other via twilio video in the browser.  One is called the ‘expert’, and the other the ‘customer’.  The customer publishes only audio and data channels.  The expert publishes audio, video and data.

The customer and expert both connect to the room and publish their channels.  When they connect - on some connections the expert is unable to hear the customer.   The issue only arises 1 out of every 20-100 runs.

**Technology:**
The application we have built is based on the react sample code.
When the expert navigates to our page - they connect to the twilio room, then (because of the behaviour of our app and the twilio hooks) immediately independently publishes data, audio and video tracks (more or less in parallel).  A similar thing will happen on the customers machine at approximately the same time.
We have turned on twilio logging.
 
**Observation**
Whenever the customer's sound is not present - it is always accompanied by a 'Glare detected; rolling back' twilio log entry on the customer's machine.
On the expert's machine the audio initially mounts, but we quickly receive a MediaStreamTrack ended event and audio tag's srcObject active becomes false. 
The broker cannot hear any sound from the customer
We receive no further events regarding additional MediaStreams / Tracks being published/started

**Log Files**
I cannot supply any code, but have attached relevant fragments of the twilio log file in the attached excel.
 
[TwilioAudioLog.xlsx](https://github.com/twilio/twilio-video.js/files/6713552/TwilioAudioLog.xlsx)

**Software versions:**
Chrome v89 on MacOs
twilio-video.js 2.15.1
react 17.0.2

**Comment by charliesantos** [comment]
Hi @EdGallimore , thanks for submitting. Can you please update to latest chrome version and post logs in a .txt or .log format?

**Comment by EdGallimore** [comment]
Hello, @charliesantos - I have verified that the issue still occurs in the latest version of chrome (v91)

I have attached text file versions of the log files below
[customer.txt](https://github.com/twilio/twilio-video.js/files/6759898/customer.txt)
[expert.txt](https://github.com/twilio/twilio-video.js/files/6759899/expert.txt)

Thanks
